### PR TITLE
Adding Logging Functionality to Proxy

### PR DESCRIPTION
**Pull request by kb1lqc**
Per #108 I implemented a method of saving data to an SQLite3 database. Largely this will be for testing Telemetry and APRS functionality. I used some methodology discussed with @b00klegger but have minimized the code additions and overall footprint of this code to remain in scope with **only logging data**. We will use separate efforts to actually test the code. This code is needed now to help flesh out how we are automating testing.

Essentially this code is extremely straightforward. I'd suggest stepping through the commits from beginning to end to see the thought process.

When enabled in `proxy.ini`, logging is performed to `log.db` in the proxy folder.

# Database Format
The database is setup in a simplistic manner with a `KeyID` that is unique and auto-increments. Then the Faraday `Port` is saved along with actual `BASE64` data. Lastly, every row is given a computer generated `EPOCH` timestamp indicating when the data was inserted into the database.

# Configuration
A simple boolean value in proxy.ini was added to control logging.

# Logging in Action
As shown below, logging is easily seen with DB Browser for SQLite
proxylog

### CODE REVIEW

File: Proxy/proxy.py [hunk 1]
```diff
@@ -71,10 +72,20 @@ def uart_worker(modem, getDicts, units):
...
+
+                            # Check for Proxy logging and save to SQL if true
+                            if log:
+                                item["port"] = port
+                                sqlInsert(item)
                         except:
                             getDicts[unit][port] = deque([], maxlen=100)
                             getDicts[unit][port].append(item)
 
+                            # Check for Proxy logging and save to SQL if true
```
Comment (reillyeon) [inline Proxy/proxy.py]: If we log `item` in either case why not move this to after the try/except block.

File: Proxy/proxy.py [hunk 2]
```diff
@@ -351,10 +362,106 @@ def callsign2COM():
...
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            return False
+
+    return True
+
+
+def sqlInsert(data):
+    """
+    Takes in a data tuple and inserts int into the Proxy SQLite table
```
Comment (reillyeon) [inline Proxy/proxy.py]: int -> it

File: Proxy/proxy.py [hunk 3]
```diff
@@ -351,10 +362,106 @@ def callsign2COM():
...
+        try:
+            conn = sqlite3.connect(db)
+
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            return False
+
+        # Connect to database, create SQL query, execute query, and close database
+        try:
+            conn = sqlite3.connect(db)
```
Comment (reillyeon) [inline Proxy/proxy.py]: `conn` was already opened in the try/except block above.

File: Proxy/proxy.py [hunk 4]
```diff
@@ -351,10 +362,106 @@ def callsign2COM():
...
+            # Use connection as context manager to rollback automatically if error
+            with conn:
+                conn.execute(sql,sqlparameters)
+
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            conn.rollback()
```
Comment (reillyeon) [inline Proxy/proxy.py]: Using `conn` as a context manager will automatically call `rollback()` so it is not necessary here.

**Review by reillyeon — DISMISSED** [review]

File: Proxy/proxy.py (hunk 1 above)
Comment (kb1lqc) [reply Proxy/proxy.py]: Good point. Fixed

File: Proxy/proxy.py (hunk 4 above)
Comment (kb1lqc) [reply Proxy/proxy.py]: Good catch! [Per the docs](https://docs.python.org/2/library/sqlite3.html#using-the-connection-as-a-context-manager). Removed

File: Proxy/proxy.py (hunk 3 above)
Comment (kb1lqc) [reply Proxy/proxy.py]: Removed. Good catch!

**Review by kb1lqd — APPROVED** [review]
Looks good.

**Pull request merged**